public/api/updatePortfolioActivitySettings.php

```diff
@@ -88,15 +74,28 @@
 }
 
 if ($success) {
-    // $sql = "
-    //     UPDATE course_content
-    //     SET label = '$label', 
-    //     imagePath = '$imagePath',
-    //     isPublic = '$isPublic',
-    //     learningOutcomes = '$learningOutcomes'
-    //     WHERE doenetId = '$doenetId'
-    //     AND courseId = '$portfolioCourseId'
-    //     ";
+
+    if ($dbIsPublic == '0' && $isPublic == '1'){
```

Comment:
Do we want to be disabling userCanViewSource when something goes back to private?

We can consolidate these two queries into one if we either want to move it back to userCanViewSource = 0, or if we don't want to touch that property we could add it to the query above that you used to grab dbIsPublic, and then just pass the current value in the db into this query.

```
if($dbIsPublic != $isPublic) {
        
	$userCanViewSource = 1;
	if (!$isPublic) {
            //if we are moving back to private don't change userCanViewSource
       	    $userCanViewSource = $dbIsUserCanViewSource
        }
        //Changing from public to private or vice versa
        $sql = "
        UPDATE course_content
        SET isPublic = '$isPublic',
        userCanViewSource = '$userCanViewSource',
        addToPrivatePortfolioDate = CONVERT_TZ(NOW(), @@session.time_zone, '+00:00')
        WHERE doenetId = '$doenetId'
        ";
        $conn->query($sql);
    }
```

If the answer to the first question is yes, this could be simplified to just pass $isPublic to set both fields.